src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -392,6 +394,54 @@ internal sealed override ObsoleteAttributeData ObsoleteAttributeData
             }
         }
 
+#nullable enable
+        internal sealed override UnmanagedCallersOnlyAttributeData? UnmanagedCallersOnlyAttributeData
+        {
+            get
+            {
+                if (syntaxReferenceOpt is null)
+                {
+                    // no references -> no attributes
+                    return null;
+                }
+
+                var lazyCustomAttributesBag = _lazyCustomAttributesBag;
+                if (lazyCustomAttributesBag is null)
+                {
+                    return UnmanagedCallersOnlyAttributeData.Uninitialized;
+                }
+
+                if (lazyCustomAttributesBag.IsDecodedWellKnownAttributeDataComputed)
+                {
+                    var data = (MethodWellKnownAttributeData?)lazyCustomAttributesBag.DecodedWellKnownAttributeData;
+#if DEBUG // Can remove ifdefs and replace with Conditional after https://github.com/dotnet/roslyn/issues/47463 is fixed
+                    verifyDataConsistent((CommonMethodEarlyWellKnownAttributeData?)lazyCustomAttributesBag.EarlyDecodedWellKnownAttributeData, data);
+#endif
+                    return data?.UnmanagedCallersOnlyAttributeData;
+                }
+
+                if (lazyCustomAttributesBag.IsEarlyDecodedWellKnownAttributeDataComputed == true)
+                {
+                    var data = (CommonMethodEarlyWellKnownAttributeData?)lazyCustomAttributesBag.EarlyDecodedWellKnownAttributeData;
+                    return data?.UnmanagedCallersOnlyAttributeDataPresent == true
+                        ? UnmanagedCallersOnlyAttributeData.AttributePresentDataNotBound
+                        : null;
+                }
+
+                return UnmanagedCallersOnlyAttributeData.Uninitialized;
+
+
+#if DEBUG // Can remove ifdefs and replace with Conditional after https://github.com/dotnet/roslyn/issues/47463 is fixed
+                static void verifyDataConsistent(CommonMethodEarlyWellKnownAttributeData? earlyData, MethodWellKnownAttributeData? lateData)
```

Comment:
>verifyDataConsistent [](start = 28, length = 20)

nit: could inline the assertion to avoid the ifdef

Reply:
With the cast, I felt it was too cluttered to be readable.

---
In reply to: [483847753](https://github.com/dotnet/roslyn/pull/47311#discussion_r483847753) [](ancestors = 483847753)